src/Resources/ResourceManager/Implementation/Providers/RegisterAzureProviderManagementGroupCmdlet.cs

```diff
@@ -0,0 +1,55 @@
+// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+
+namespace Microsoft.Azure.Commands.ResourceManager.Cmdlets.Implementation
+{
+    using Microsoft.Azure.Commands.ResourceManager.Cmdlets.SdkModels;
+    using System.Collections.Generic;
+    using System.Management.Automation;
+    using ProjectResources = Microsoft.Azure.Commands.ResourceManager.Cmdlets.Properties.Resources;
+
+    /// <summary>
+    /// "Registers a management group with a resource provider.
+    /// </summary>
+    [Cmdlet("RegisterAtManagementGroupScope", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "ResourceProvider", SupportsShouldProcess = true), OutputType(typeof(PSResourceProvider))]
```

Comment:
1. The verb is invalid and the noun is conflict with `Register-AzResourceProvider`.